src/initRules.js

```diff
@@ -175,17 +171,34 @@ module.exports = function(
           var timeoutId;
 
           return new Promise(function(resolve, reject) {
-            timeoutId = setTimeout(function() {
-              reject(
-                'A timeout occurred because the action took longer than ' +
-                  PROMISE_TIMEOUT / 1000 +
-                  ' seconds to complete. '
-              );
-            }, PROMISE_TIMEOUT);
-
-            Promise.resolve(
-              executeDelegateModule(action, syntheticEvent, [syntheticEvent])
-            ).then(resolve, reject);
+            var promiseTimeout = action.settings.timeout;
+
+            if (
+              promiseTimeout &&
+              (isNaN(promiseTimeout) || promiseTimeout < 0)
+            ) {
+              reject(new Error('Action timeout is not correctly defined.'));
```

Comment:
Since you don't return from the function, the action will still be executed (the same goes for when a condition's timeout is invalid). Is this intentional? I'm not sure what I would expect as a customer TBH. I don't see it asserted in the test either way, but I might have just overlooked it.

Reply:
It's a mistake from my part. Probably we should return in both cases.

Reply:
I've removed the code and the tests related timeout.